scipy/ndimage/tests/test_filters.py

```diff
@@ -1213,6 +1233,29 @@ def test_maximum_filter09(self):
                                    [7, 9, 8, 9, 7],
                                    [8, 8, 8, 7, 7]], output)
 
+    @pytest.mark.parametrize(
+        'axes', tuple(itertools.combinations(range(-3, 3), 2))
+    )
+    @pytest.mark.parametrize(
+        'filter_func', [ndimage.minimum_filter, ndimage.maximum_filter]
+    )
+    def test_minmax_nonseparable_axes(self, filter_func, axes):
+        array = numpy.arange(6 * 8 * 12, dtype=numpy.float32).reshape(6, 8, 12)
+        # use 2D triangular footprint because it is non-separable
+        footprint = numpy.tri(5)
+        axes = tuple(ax % array.ndim for ax in axes)
```

Comment:
Note that `axes` will always become positive here. I'd suggest changing this like we did above so `axes` can have negative elements.

Reply:
now addressed for this and `test_filter_axes` in [bde038b](https://github.com/scipy/scipy/pull/18261/commits/bde038b960d7ec4dbe97ce903d86ea2a1debd7fd)